extension/js/common/ui/key-import-ui.ts

```diff
@@ -105,18 +105,18 @@ export class KeyImportUi {
     const attach = new AttachmentUI(() => Promise.resolve({ count: 100, size: 1024 * 1024, size_mb: 1 }));
     attach.initAttDialog('fineuploader', 'fineuploader_button', {
       attAdded: async file => {
-        let prv: OpenPGP.key.Key | undefined;
+        let prv: Key | undefined;
         const utf = file.getData().toUtfStr();
         if (utf.includes(PgpArmor.headers('privateKey').begin)) {
           const firstPrv = MsgBlockParser.detectBlocks(utf).blocks.filter(b => b.type === 'privateKey')[0];
           if (firstPrv) { // filter out all content except for the first encountered private key (GPGKeychain compatibility)
-            prv = (await opgp.key.readArmored(firstPrv.content.toString())).keys[0];
+            prv = (await KeyUtil.parse(firstPrv.content.toString()));
           }
         } else {
-          prv = (await opgp.key.read(file.getData())).keys[0];
+          prv = (await KeyUtil.parseBinary(file.getData(), String(prompt('Key password'))))[0];
```

Comment:
This logic here needs to be inverted. That is node-forge requires good password but when clicking on "import file" we don't have the password field shown yet.

I think we need to do only small level of sanity checks for X.509 keys here as it seems that's what the function is doing here (reads the key and immediately armors it discarding the object) and then do the full parse in `checkPrv`.